.pre-commit-config.yaml

```diff
@@ -108,26 +108,24 @@ repos:
           .* See the License for the specific language governing permissions and
           .* limitations under the License.
         language: pygrep
-        files: \.js$|\.cjs$|\.mjs$|\.ts$|\.py$
+        types: [javascript, ts, python]
         exclude: __init__\.py$
         args: [--negate, --multiline]
       - id: node-self
         name: check nodejs `self.`
         entry: '\bself\.'
         language: pygrep
-        files: \.ts$
+        types: [ts]
       # ESLint is the slowest hook, so keep it at the bottom.
       - id: eslint
         name: eslint
         entry: 'npx eslint --fix'
         language: system
         verbose: true
-        files: \.[jt]s$
-        types: [file]
+        types: [javascript, ts]
       # ESLint may update code style (formatting) so keep prettier last.
       - id: prettier
         name: prettier
-        entry: 'npx prettier --check .'
+        entry: 'npx prettier --write --ignore-unknown --check .'
```

Comment:
Do we need the `.` at the end? I think that will run all files Prettier can format in the repo? We may if we want to format `.yml`, `.json` etc.

Reply:
No. Removed.

Reply:
Yes, that's OK. Markdown, YAML, etc. Whatever prettier supports.